pkg/secrets/secrets.go

```diff
@@ -127,9 +126,33 @@ func getMountsMap(path string) (string, string, error) {
 	return "", "", errors.Errorf("unable to get host and container dir")
 }
 
-// SecretMounts copies the contents of host directory to container directory
+// SecretMounts copies, adds, and mounts the secrets to the container root filesystem
+func SecretMounts(mountLabel, containerWorkingDir string) []rspec.Mount {
+	var secretMounts []rspec.Mount
+	// Add secrets from paths given in the mounts.conf files
+	for _, file := range []string{OverrideMountsFile, DefaultMountsFile} {
+		mounts, err := addSecretsFromMountsFile(file, mountLabel, containerWorkingDir)
+		if err != nil {
+			logrus.Warnf("error mounting secrets, skipping: %v", err)
+		}
+		secretMounts = append(secretMounts, mounts...)
+	}
+
+	// Add FIPS mode secret if /etc/system-fips exists on the host
+	_, err := os.Stat("/etc/system-fips")
```

Comment:
If I can't check if the /etc/system-fips file exists. then I can not tell whether to put the system into fips mode, but we don't even exist.  We should at least logrus.Errorf() this fact,

Reply:
If it's os.IsNotExist(), the system is not in FIPS mode, we should just debugf. If it's an actual access error, I think that's worthy of an Errorf (or returning an errors.Wrapf?)

Reply:
Fixed.

Reply:
If you can't search /etc then you don't know if the system is in fips mode, so we really should not continue.  (Highly unlikely).